autogpt_platform/backend/backend/util/service.py

```diff
@@ -170,31 +185,53 @@ def __init__(self):
             logger.debug(f"Successfully connected to service [{service_name}]")
 
         def __getattr__(self, name: str) -> Callable[..., Any]:
-            return getattr(self.proxy, name)
+            res = getattr(self.proxy, name)
+            return res
 
     return cast(AS, DynamicClient())
 
 
 # --------- UTILITIES --------- #
 
+builtin_types = [*vars(builtins).values(), NoneType, Enum]
+
 
 def _pydantic_models_from_type_annotation(annotation) -> Iterator[type[BaseModel]]:
     # Peel Annotated parameters
     if (origin := get_origin(annotation)) and origin is Annotated:
         annotation = get_args(annotation)[0]
 
-    if origin := get_origin(annotation):
-        if origin is UnionType:
-            types = get_args(annotation)
-        else:
-            types = [origin]
+    origin = get_origin(annotation)
+    args = get_args(annotation)
+
+    if origin in (
+        Union,
+        UnionType,
+        list,
+        List,
+        tuple,
+        Tuple,
+        set,
+        Set,
+        frozenset,
+        FrozenSet,
+    ):
+        for arg in args:
+            yield from _pydantic_models_from_type_annotation(arg)
+    elif origin in (dict, Dict):
+        key_type, value_type = args
+        yield from _pydantic_models_from_type_annotation(key_type)
+        yield from _pydantic_models_from_type_annotation(value_type)
     else:
-        types = [annotation]
+        annotype = annotation if origin is None else origin
 
-    for annotype in types:
+        # Exclude generic types and aliases
         if (
             annotype is not None
-            and not hasattr(typing, annotype.__name__)  # avoid generics and aliases
-            and issubclass(annotype, BaseModel)
+            and not hasattr(typing, getattr(annotype, "__name__", ""))
+            and isinstance(annotype, type)
         ):
-            yield annotype
+            if issubclass(annotype, BaseModel):
+                yield annotype
+            elif annotype not in builtin_types and not issubclass(annotype, Enum):
```

Comment:
I added explicit exception when unrecognized type is used on the exposed function.